src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -623,8 +624,10 @@ internal Boolean LoadLocalHeaderExtraFieldAndCompressedBytesIfNeeded()
                 _compressedBytes = new Byte[(_compressedSize / MaxSingleBufferSize) + 1][];
                 for (int i = 0; i < _compressedBytes.Length - 1; i++)
                 {
-                    _compressedBytes[i] = new Byte[MaxSingleBufferSize];
+                    _compressedBytes[i] = ArrayPool<byte>.Shared.Rent(MaxSingleBufferSize);
```

Comment:
> we would need to add Dispose

That would need to be added to the full framework as well.  And this is a public type, which means code beyond these compression types would now be using a finalizable object without disposing it.

My $.02: I would suggest either we don't use the pool at all, or if we think the majority of usage will hit the code paths that call Return, we accept that some buffers may not be returned.